src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala

```diff
@@ -24,6 +24,61 @@ trait InlineErrorChecker {
     recursivePreds
   }
 
+  /**
+    * For each predicate id, find its associated body in the program. If the predicate is nested,
+    * print a warning to the console that it shall not be inlined. Return a set of predicates that
+    * are recursive
+    *
+    * @param predicateIds the ids of the predicates we want to inline.
+    * @param program the program for which we are performing predicate inlining on.
+    * @return the set of nested predicates.
+    */
+  def checkNestedPreds(predicateIds: Set[String], program: Program): Set[Predicate] = {
+    val predicates = predicateIds.map(program.findPredicate)
+    val nestedPreds = predicates.filter {
+      case Predicate(name, _, _) => isNestedPred(name, predicates)
+    }
+    nestedPreds.foreach {
+      case Predicate(name, _, _) => println(s"Predicate: `$name` is a nested predicate. Will not be inlined")
```

Comment:
Minor thing: I'd it to be a full sentence (e.g. `... is a nested predicate and will not be inlined.`)

Reply:
Resolved